Sources/Basics/FileSystem/Path+Extensions.swift

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import struct TSCBasic.AbsolutePath
+
+extension AbsolutePath {
+    public func appending(_ component: String) -> AbsolutePath {
+        self.appending(component: component)
+    }
+
+    public func appending(_ components: String...) -> AbsolutePath {
+        self.appending(components: components)
+    }
+
+    public func withExtension(_ extension: String) -> AbsolutePath {
+        guard !self.isRoot else { return self }
+        let `extension` = `extension`.spm_dropPrefix(".")
+        return self.parentDirectory.appending("\(basename).\(`extension`)")
+    }
+
+    public func basenameWithoutAnyExtension() -> String {
```

Comment:
`Any` feels redundant to me here, but it could be plural to indicate that multiple extensions can be removed IIUC
```suggestion
    public func basenameWithoutExtensions() -> String {
```

Reply:
I just moved this function from somewhere else. I think it was originally written by @abertelrud  he may have had a read to name it this way?

Reply:
actually I think we already have `basenameWithoutExtension`, so this is a more nuanced version which strips multiple ones?